### PR TITLE
Private messages

### PR DESCRIPTION
**Pull request by Crysicia**
Idk what's happening...

### CODE REVIEW

**Review by gmorange42 — APPROVED** [review]
Great job !

**Review by lambrette — APPROVED** [review]
Good job,  remove your comments and it will be perfect.

**Review by gmorange42 — APPROVED** [review]
Where are the comments ? (Joke) Good job, it works perfectly !

File: backend/src/chat/services/chatroom.socket.service.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+  async leaveChatroom(
+    client: Socket,
+    server: Server,
+    payload: { chatroomId: string }
+  ) {
+    const userId = client['decoded'].sub;
+    const chatroom = await this.chatroom.findOne(payload.chatroomId);
+
+    if (chatroom.type !== RoomType.ONE_TO_ONE) {
```
Comment (tsiguenz) [inline backend/src/chat/services/chatroom.socket.service.ts:80]: In front we can leave private message, is it working ?

File: backend/src/private_message/private_message.service.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      include: { users: true }
+    });
+  }
+
+  async findOrCreate(
+    firstUserId: string,
+    secondUserId: string
+  ): Promise<ChatRoom> {
+    const chatroom = await this.findOne(firstUserId, secondUserId);
+    console.log(chatroom);
```
Comment (tsiguenz) [inline backend/src/private_message/private_message.service.ts:44]: remove log pls

File: backend/src/private_message/private_message.service.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  async create(firstUserId: string, secondUserId: string): Promise<ChatRoom> {
+    const firstUser = await this.prisma.user.findUnique({
+      where: { id: firstUserId }
+    });
+    const secondUser = await this.prisma.user.findUnique({
+      where: { id: secondUserId }
+    });
+
+    const roomName = `(${firstUser.nickname} - ${secondUser.nickname}) Private message`;
```
Comment (tsiguenz) [inline backend/src/private_message/private_message.service.ts:59]: Is this fine ? What happens if we try to create a room with the same name ? Before or after private messages rooms are created ? If it's protected it's not better to print the name of the user we are talking with in front ?

File: frontend/src/services/chat.service.ts [hunk 4]
```diff
@@ -82,23 +120,38 @@ class ChatService {
...
   }
 
   kickUser(userId: string, chatroomId: string) {
-    this.socketService.send(events.KICK_USER, { userId, chatroomId });
+    this.socketService.send(events.CHATROOM_KICK, { userId, chatroomId });
+  }
+
+  createOneToOne(firstUserId: string, secondUserId: string) {
+    this.socketService.send(events.CHATROOM_CREATE, {
+      name: 'SuperDuperChat',
```
Comment (tsiguenz) [inline frontend/src/services/chat.service.ts:144]: what is this ?

**Review by tsiguenz — CHANGES_REQUESTED** [review]
Great job but some things need to be improved in the chat. Private messages seems to work fine. I know its the main purpose of this PR 👀

**Review by tsiguenz — APPROVED** [review]
thanks

File: frontend/src/services/chat.service.ts (hunk 4 above)
Comment (Crysicia) [reply frontend/src/services/chat.service.ts:144]: Baby don't hurt me

File: backend/src/chat/services/chatroom.socket.service.ts (hunk 1 above)
Comment (Crysicia) [reply backend/src/chat/services/chatroom.socket.service.ts:80]: Done

File: backend/src/private_message/private_message.service.ts (hunk 3 above)
Comment (Crysicia) [reply backend/src/private_message/private_message.service.ts]: Fixed

**Review by gmorange42 — APPROVED** [review]
Nice explanations, looks good for me.

**Pull request merged**